python/mlc_chat/compiler/compiler_pass/cublas_dispatch.py

```diff
@@ -0,0 +1,47 @@
+"""A compiler pass that dispatches patterns to CUBLAS."""
+import tvm
+import os
+from tvm import IRModule, relax
+from tvm.relax.backend import get_patterns_with_prefix
+import tvm.relax.backend.contrib.cublas as _cublas
+from ..compile import CompileArgs
+
+
+@tvm.transform.module_pass(opt_level=0, name="CublasDispatch")
+class CublasDispatch:  # pylint: disable=too-few-public-methods
+    """A compiler pass that dispatches patterns to CUBLAS."""
+
+    def __init__(self, args: CompileArgs) -> None:
+        self.args = args
+
+    def transform_module(self, mod: IRModule, _ctx: tvm.transform.PassContext) -> IRModule:
+        """IRModule-level transformation"""
+        if self.args.target.kind.name != "cuda":
+            return mod
+
+        patterns = []
+
+        has_cublas = tvm.get_global_func("relax.ext.cublas", True)
+
+        if (
+            has_cublas
+            and self.args.quantization.name in ("q0f16", "q0f32")
```

Comment:
curious why it's only enabled for fp16 or fp32? even for w4a16, when attention is not offloaded, there are batched gemvs that cublas supports